components/usage/pkg/apiv1/billing.go

```diff
@@ -165,6 +177,61 @@ func (s *BillingService) CreateStripeCustomer(ctx context.Context, req *v1.Creat
 	}, nil
 }
 
+func (s *BillingService) CreateStripeSubscription(ctx context.Context, req *v1.CreateStripeSubscriptionRequest) (*v1.CreateStripeSubscriptionResponse, error) {
+	attributionID, err := db.ParseAttributionID(req.GetAttributionId())
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Invalid attribution ID %s", attributionID)
+	}
+
+	if req.CustomerId == "" {
+		return nil, status.Error(codes.InvalidArgument, "Invalid customer ID")
+	}
+
+	_, err = s.stripeClient.SetDefaultPaymentForCustomer(ctx, req.CustomerId, req.SetupIntentId)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to set default payment for customer ID %s", req.CustomerId)
+	}
+
+	stripeCustomer, err := s.stripeClient.GetCustomer(ctx, req.CustomerId)
+	if err != nil {
+		return nil, err
+	}
+
+	priceIdentifier := getPriceIdentifier(attributionID, stripeCustomer, s)
+
+	subscription, err := s.stripeClient.CreateSubscription(ctx, req.CustomerId, priceIdentifier)
```

Comment:
Nit: Please remember to also port all the other subscription creation parameters 😇 

https://github.com/gitpod-io/gitpod/blob/7532f3c15fbad543c0efbaa4633418ef609f599c/components/server/ee/src/user/stripe-service.ts#L161-L166

- `automatic_tax` is important for taxation & legal reasons
- `billing_cycle_anchor` is important so that billing period always go from beginning of the month to end of the month (and not e.g. from October 17 @ 14:42 to November 17 @ 14:42)

Reply:
Internal thread on `billing_cycle_anchor`[[1](https://gitpod.slack.com/archives/C02EN94AEPL/p1666863399799159)]